modules/custom/az_finder/src/Form/AZFinderSettingsForm.php

```diff
@@ -0,0 +1,373 @@
+          ":view_id" => $view_id,
+          ":display_id" => $display_id,
+        ]),
+        '#open' => FALSE,
+        '#description' => $this->t('Overrides are grouped by vocabulary. Each vocabulary can have its own settings for how term ID widgets behave when they have child terms.'),
+        '#tree' => TRUE,
+      ];
+      $config_name = "az_finder.tid_widget.{$view_id}.{$display_id}";
+      $config = $this->config($config_name);
+      $vocabulary_ids = $this->azFinderVocabulary->getVocabularyIdsForFilter($view_id, $display_id, 'taxonomy_index_tid');
+
+      foreach ($vocabulary_ids as $vocabulary_id) {
+        $this->azFinderVocabulary->addTermsTable(
+              $form['az_finder_tid_widget']['overrides'][$key]['vocabularies'][$vocabulary_id],
+              $vocabulary_id,
+              $view_id,
+              $display_id
+          );
+
+      }
+
+      $form['az_finder_tid_widget']['overrides'][$key]['delete'] = [
+        '#type' => 'submit',
+        '#value' => $this->t('Delete'),
+        '#ajax' => [
+          'callback' => '::ajaxDeleteOverride',
+          'wrapper' => 'js-overrides-container',
+          'effect' => 'fade',
+        ],
+        '#name' => 'delete-' . $key,
+        '#submit' => ['::submitDeleteOverride'],
+      ];
+    }
+    $overrides = $form_state->get('overrides') ?? [];
+    $form_state->set('overrides', $overrides);
+    $form_state->setRebuild(TRUE);
+    return $form['az_finder_tid_widget']['overrides'][$key];
+  }
+
+  /**
+   * Ajax callback for the delete button.
+   */
+  public function ajaxDeleteOverride(array &$form, FormStateInterface $form_state) {
+    $triggering_element = $form_state->getTriggeringElement();
+    $button_name = $triggering_element['#name'];
+    $key = str_replace('delete-', '', $button_name);
+
+    $overrides = $form_state->get('overrides') ?? [];
+    unset($overrides[$key]);
+    $form_state->set('overrides', $overrides);
+
+    // Rebuild the form.
+    $form_state->setRebuild(TRUE);
+
+    return $form['az_finder_tid_widget']['overrides'];
+  }
+
+  /**
+   * Separate submit handler for the delete button.
+   *
+   * @param array $form
+   *   The form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state object.
+   */
+  public function submitDeleteOverride(array &$form, FormStateInterface $form_state) {
+    $triggering_element = $form_state->getTriggeringElement();
+    $button_name = $triggering_element['#name'];
+    $key = str_replace('delete-', '', $button_name);
+    [$view_id, $display_id] = explode(':', $key);
+    $config_name = "az_finder.tid_widget.$view_id.$display_id";
+    $config = $this->config($config_name);
+    if ($config && !$config->isNew()) {
+      $editable_config = $this->configFactory->getEditable($config_name);
+      $editable_config->delete();
+    }
+    // Update the overrides in form state.
+    $overrides = $form_state->get('overrides') ?? [];
+    unset($overrides[$key]);
```

Comment:
Need to delete the config so the rebuild form doesn't find it when building the overrides section.
```suggestion
    unset($overrides[$key]);
    $this->config('az_finder.tid_widget.' . $key)->delete();
```